### PR TITLE
cmd/atlasci/ci: support loading migration changes

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: cmd/atlasci/internal/ci/ci.go [hunk 1]
```diff
@@ -38,15 +43,19 @@ type (
 		migrate.Scanner
 	}
 
-	// Change wraps a schema.Change and adds metadata to it.
-	Change struct {
-		schema.Change
-		// Generated determines if this file was solely generated by Atlas and not edited later on by a human.
-		Generated bool
+	// File represents a parsed version of a migration file.
```
Comment (masseelch) [inline cmd/atlasci/internal/ci/ci.go:46]: :+1:

File: cmd/atlasci/internal/ci/ci.go [hunk 2]
```diff
@@ -182,3 +191,69 @@ func treeObject(r *git.Repository, n plumbing.ReferenceName) (*object.Tree, erro
 	}
 	return t, nil
 }
+
+// DevLoader implements the ChangesLoader interface using a dev-driver.
+type DevLoader struct {
+	// Dev environment used as a sandbox instantiated to the starting point (e.g. base branch).
+	Dev *sqlclient.Client
+	// Scan is used for scanning the migration directory.
+	Scan migrate.Scanner
```
Comment (masseelch) [inline cmd/atlasci/internal/ci/ci.go:200]: Why exported fields?

File: sql/sqlite/driver.go [hunk 3]
```diff
@@ -49,7 +49,12 @@ func init() {
 		sqlclient.RegisterCodec(MarshalHCL, EvalHCL),
 		sqlclient.RegisterFlavours("sqlite"),
 		sqlclient.RegisterURLParser(sqlclient.URLParserFunc(func(u *url.URL) *sqlclient.URL {
-			return &sqlclient.URL{URL: u, DSN: strings.TrimPrefix(u.String(), u.Scheme+"://"), Schema: mainFile}
+			uc := &sqlclient.URL{URL: u, DSN: strings.TrimPrefix(u.String(), u.Scheme+"://"), Schema: mainFile}
+			if mode := u.Query().Get("mode"); mode == "memory" {
+				// The "file:" prefix in mandatory for memory modes.
```
Comment (masseelch) [inline sql/sqlite/driver.go:54]:
```suggestion
				// The "file:" prefix is mandatory for memory modes.
```

**Review by masseelch — DISMISSED** [review]
:rocket:

File: cmd/atlasci/internal/ci/ci.go (hunk 2 above)
Comment (a8m) [reply cmd/atlasci/internal/ci/ci.go:200]: Avoid creating a constructor at this stage. I may add one at some point.

**Pull request merged**